jormungandr/src/blockchain/branch.rs

```diff
@@ -97,7 +97,7 @@ impl Branch {
         }
     }
 
-    pub fn get_ref<E>(&self) -> impl Future<Item = Arc<Ref>, Error = E> {
+    pub fn get_ref(&self) -> impl Future<Item = Arc<Ref>, Error = Infallible> {
```

Comment:
Please revert. The infaillible type of for convertion. Having a type parameter to say any error type make it easier to use and avoid the pattern matching to map an unreachable.

This is an issue of the legacy future.

Reply:
I see, but then it makes type inference fail in some other use sites, so you have to uglify the code with type annotations. Pick your poison.

If pattern matching is too ugly, we can use this little [crate](https://crates.io/crates/unwrap-infallible).

Reply:
I still think this is the wrong approach. Adding a type to fix the problem of the `Future` API is the wrong approach. Especially that we will still need to remove it down the line with the port to new Future API.

Reply:
`Result<_, Infallible>` is exactly the new `Future` API